lib/packages/docutils/rstgen.nim

```diff
@@ -1136,19 +1138,45 @@ proc renderAdmonition(d: PDoc, n: PRstNode, result: var string) =
         "$1\n\\end{mdframed}\n",
       result)
 
+const anchorLink = """<a class="nimanchor" id="$2" href="#$2">🔗</a>"""
+
+proc computeAnchorGen(n: PRstNode) =
+  var anchorLast = "ROOT"
+  var idLast = 0
+  proc impl(n: PRstNode) =
+    if n == nil: return
+    n.anchorGen = "\0" # special char to indicate we've visited this node.
```

Comment:
Why is this needed? It's supposed to be a strict tree structure or at least an acyclic graph.

Reply:
it's needed because I'm processing the whole tree/DAG upon 1st call to `renderRstToOut`, and because `renderRstToOut` is reentrant. This sentinel avoids needs for a separate flag.
Without this logic, you can see it doesn't work:
nim rst2html -r doc/contributing.rst
the anchor next to `separate test files` would be `#-ROOT-2` instead of `#-writing-tests-3`